File: Arbitration/MPArbitration/Model/PlaceOfServiceCode.cs
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace MPArbitration.Model
{
    [Index(nameof(CodeNumber), IsUnique = true)]
    public class PlaceOfServiceCode
    {
        [JsonPropertyName("id")]
        public int Id { get; set; } = 0;

        [Required]
        [StringLength(100)]
        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [StringLength(1000)]
        [JsonPropertyName("description")]
        public string Description { get; set; } = string.Empty;

        [Required]
        [StringLength(4)]  // allow for future expansion
        [JsonPropertyName("codeNumber")]
        public string CodeNumber { get; set; } = string.Empty;

        [JsonPropertyName("effectiveDate")]
        public DateTime? EffectiveDate { get; set; } = null;
    }
}

File: Arbitration/MPArbitration/ClientApp/src/app/model/place-of-service-code.ts
export class PlaceOfServiceCode {
    public id = 0;
    public codeNumber = '';
    public description = '';
    public effectiveDate:Date|undefined;
    public name = '';

    constructor(obj?:any){
        if(!obj)
            return;

        Object.assign(this,JSON.parse(JSON.stringify(obj)));
        if(this.effectiveDate)
            this.effectiveDate = new Date(obj.effectiveDate);
    }
}